projects/libs/core-library-typescript/src/common-library/block-utils.ts

```diff
@@ -0,0 +1,38 @@
+
+import { types, logger } from ".";
+import { createHash } from "crypto";
+
+export namespace BlockUtils {
+  export function calculateBlockHash(blockComponents: {header: types.BlockHeader, body: types.BlockBody}): Buffer {
+    const hash = createHash("sha256");
```

Comment:
It's common to do a double-sha256 here (i.e., `hash = sha256(sha256(block))`) as a "better safe than sorry" mitigation against a similar attack on sha1. Alternatively, we can do sha3 (keccak) once instead.

Reply:
It's more a placeholder until the hashing scheme is finalized